src/Symfony/Component/PropertyAccess/PropertyAccessor.php

```diff
@@ -651,6 +651,8 @@ private function writeProperty($zval, $property, $value)
             // fatal error.
 
             $object->$property = $value;
+        } elseif (self::ACCESS_TYPE_NOT_FOUND === $access[self::ACCESS_TYPE]) {
+            throw new NoSuchPropertyException(sprintf('Could not determine access type for property "%s".', $property));
         } elseif (self::ACCESS_TYPE_MAGIC === $access[self::ACCESS_TYPE]) {
```

Comment:
That would't be changed, I mean only `else/elseif` blocks directly adjacent after `return` or `throw` statments (in this case, `if (self::ACCESS_TYPE_MAGIC === $access[self::ACCESS_TYPE])`)